aidial_adapter_openai/utils/parsers.py

```diff
@@ -52,32 +52,45 @@ def get_client(self, params: OpenAIParams) -> AsyncOpenAI:
         )
 
 
-class EndpointParser(BaseModel):
-    name: str
-
-    def parse(self, endpoint: str) -> AzureOpenAIEndpoint | OpenAIEndpoint:
-        match = re.search(
-            f"(.+?)/openai/deployments/(.+?)/{self.name}", endpoint
+def _parse_endpoint(
+    name, endpoint
+) -> AzureOpenAIEndpoint | OpenAIEndpoint | None:
+    if azure_match := re.search(
+        f"(.+?)/openai/deployments/(.+?)/{name}", endpoint
+    ):
+        return AzureOpenAIEndpoint(
+            azure_endpoint=azure_match[1],
+            azure_deployment=azure_match[2],
         )
 
-        if match:
-            return AzureOpenAIEndpoint(
-                azure_endpoint=match[1],
-                azure_deployment=match[2],
-            )
+    if openai_match := re.search(f"(.+?)/{name}", endpoint):
+        return OpenAIEndpoint(base_url=openai_match[1])
 
-        match = re.search(f"(.+?)/{self.name}", endpoint)
 
-        if match:
-            return OpenAIEndpoint(base_url=match[1])
+class EndpointParser(BaseModel):
+    name: str
 
+    def parse(self, endpoint: str) -> AzureOpenAIEndpoint | OpenAIEndpoint:
+        if result := _parse_endpoint(self.name, endpoint):
+            return result
         raise DialException(
             "Invalid upstream endpoint format", 400, "invalid_request_error"
         )
 
 
+class LegacyCompletionsParser(BaseModel):
```

Comment:
It doesn't really bring anything of value that we point out here that `/completions` endpoint is **legacy**.

Moreover, it's only legacy in **OpenAI**, but many other models _(e.g. one from HuggingFace)_ do support completions mode and this might be the only mode they support. And thus there is nothing legacy about it.

Reply:
I suggest find-and-replace "legacy" -> "" throughout the project.

Reply:
Fixed, removed "legacy*"